landlab/components/snow/snow_degree_day.py

```diff
@@ -0,0 +1,300 @@
+"""Landlab component that simulates snowpack dynamics.
+
+This component simulates snowmelt process using the degree-day method.
+The code is implemented based on the TopoFlow snow component (by Scott D. Packham)
+with some adjustments.
+https://github.com/peckhams/topoflow36/blob/master/topoflow/components/snow_degree_day.py
+https://github.com/peckhams/topoflow36/blob/master/topoflow/components/snow_base.py
+
+@author: Tian Gan June 2024
+
+"""
+
+import numpy as np
+
+from landlab import Component
+
+
+class SnowDegreeDay(Component):
+    """Simulate snowmelt process using degree-day method.
+
+      The degree-day method uese c0 (degree-day coefficient),
+      T0 (degree-day threshold temperature) and T_air (air temperature)
+      to determine the snow melt rate.
+      When T_air is larger than T0, the melt process will start.
+      The melt rate is calculated as SM = (T_air -T0) * c0
```

Comment:
Suggest defining the units of SM and c0 (for example, if SM is in m/day or whatever)